code/renderers/vue3/src/render.ts

```diff
@@ -25,9 +25,18 @@ export const render: ArgsStoryFn<VueRenderer> = (props, context) => {
   return h(Component, props, createOrUpdateSlots(context));
 };
 
-let setupFunction = (_app: any) => {};
-export const setup = (fn: (app: any) => void) => {
-  setupFunction = fn;
+// set of setup functions that will be called when story is created
+const setupFunctions = new Set<(app: App, storyContext?: StoryContext<VueRenderer>) => void>();
+/** add a setup function to set that will be call when story is created a d
+ *
+ * @param fn
+ */
+export const setup = (fn: (app: App, storyContext?: StoryContext<VueRenderer>) => void) => {
+  setupFunctions.add(fn);
```

Comment:
don't we need to remove setup functions again at some point, likely during teardown?
I'm thinking that if we don't, the setup functions will pollute each other, so if you open story A with its setup function, and then story B with its setup function, then when we render story B, we will run setup both for story A and B?

Reply:
@JReinhold this was not good implementation in previous version, and it has one comment by @shilman  that we should look for better implementation 
https://github.com/storybookjs/storybook/blob/303e39407653d0d0b7e23291dc01ed806f3e5f8a/code/renderers/vue3/template/stories/preview.js

Reply:
```ts
// TODO: I'd like to be able to export rather than imperatively calling an imported function
// export const setup = (app) => {
//   app.component('GlobalButton', Button);
// };
```
so he really had the right concept which is making these functions available for other modules,  but the implementation is easy with vue, instead of exporting them you just have to add them to global Set or Array,  so at creation of Vue App you can iterate on this Set of functions ( i used Set to avoid duplication ) and run them